e2e/compatibility/consul_test.go

```diff
@@ -0,0 +1,549 @@
+// +build e2e
+
+// $ go test -run TestCompatibilty_Consul ./e2e/compatibility -tags=e2e -v
+package compatibility
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-terraform-sync/api"
+	"github.com/hashicorp/consul-terraform-sync/templates/tftmpl"
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	capi "github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-getter"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	tempDirPrefix = "tmp_"
+	resourcesDir  = "resources"
+	configFile    = "config.hcl"
+
+	nullTaskName = "null_task"
+)
+
+func TestCompatibilty_Consul(t *testing.T) {
+	// Tested only OSS GA releases for the highest patch version given a
+	// major minor version. v1.4.5 starts losing compatibility, details in
+	// comments. Theoretical compatible versions 0.1.0 GA:
+	consulVersions := []string{"1.9.4", "1.8.9", "1.7.13", "1.6.10", "1.5.3"}
```

Comment:
We should be able to programmatically determine this list for the most part, but a hard coded list is a good starting point.

Reply:
Yes! Love the idea of a programatic way to determine this list! The hard-coded list isn't great for automatically running this in Circle and would need some up-keep